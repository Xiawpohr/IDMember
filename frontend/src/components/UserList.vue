<template>
  <p>user list</p>
</template>

<script>
export default {
  props: {
    users: {
      type: Array,
      required: true,
      default: () => []
    }
  }
}
</script>
