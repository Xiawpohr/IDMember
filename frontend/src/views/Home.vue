<template>
  <div class="home">
    <h1>Discover Your Friends</h1>
    <UserList :users="users"/>
  </div>
</template>

<script>
import UserList from '@/components/UserList.vue'

export default {
  name: 'home',
  components: {
    UserList
  },
  computed: {
    users() {
      return this.$store.state.users
    }
  }
}
</script>
