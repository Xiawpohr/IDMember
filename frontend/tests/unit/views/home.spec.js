import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Home from '@/views/Home.vue'
import UserList from '@/components/UserList.vue'

const localVue = createLocalVue()
localVue.use(Vuex)
let wrapper, state, store

beforeEach(() => {
  state = {
    users: [{ name: 'Arthur' }, { name: 'Robert' }]
  }
  store = new Vuex.Store({
    state
  })
  wrapper = shallowMount(Home, {
    localVue,
    store
  })
})

it('should render the component', () => {
  expect(wrapper.html()).toMatchSnapshot()
})

it('should render UserList component', () => {
  const userList = wrapper.find(UserList)
  expect(userList.exists()).toBe(true)
})

it('should pass users prop to UserList component', () => {
  const userList = wrapper.find(UserList)
  expect(userList.vm.users).toBe(state.users)
})
