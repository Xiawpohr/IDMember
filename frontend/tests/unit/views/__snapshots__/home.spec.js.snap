// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<div class="home">
  <h1>Discover Your Friends</h1>
  <userlist-stub users="[object Object],[object Object]"></userlist-stub>
</div>
`;
